lib/internal/errors.js

```diff
@@ -85,4 +85,5 @@ module.exports = exports = {
 //
 // Note: Please try to keep these in alphabetical order
 E('ERR_ASSERTION', (msg) => msg);
+E('ERR_ARG_FUNCTION', '"block" argument must be a function');
```

Comment:
I think `ERR_INVALID_ARG_TYPE` would be more suitable for this, refs: https://github.com/nodejs/node/pull/11294/files, I believe the idea at the moment is that concurrent PRs can duplicate some of the error codes introduced in others in order to get a little bit more consistency.

Also whichever the error code would be, can you add the corresponding documentation for it? See [Documenting new errors](https://github.com/nodejs/node/blob/master/doc/guides/using-internal-errors.md#documenting-new-errors).